src/cms/templatetags/page_filters.py

```diff
@@ -19,3 +19,79 @@ def get_last_root_page(pages):
     """
     root_pages = list(filter(lambda p: not p.parent, pages))
     return root_pages[-1] if root_pages else None
+
+
+@register.simple_tag
+def get_children_in(node, pageset):
+    """
+    This tag returns the IDs of children pages of node which are in pageset.
+
+    :param node : the page
+    :type node : ~cms.models.pages.page.Page
+
+    :param pageset: The pages (all pages or pages chosen by filter)
+    :type pageset: list [ ~cms.models.pages.page.Page ]
+
+    :return: The list of IDs of children pages of node that are in pageset
+    :rtype: list [~cms.models.pages.page.Page]
+    """
+    return [child.id for child in node.children.all() if child in pageset]
+
+
+@register.simple_tag
+def get_depth_in(node, pageset):
+    """
+    This tag returns the depth of node whithin the tree/pages in pageset.
+
+    :param node : the page
+    :type node : ~cms.models.pages.page.Page
+
+    :param pageset: The pages (all pages or pages chosen by filter)
+    :type pageset: list [ ~cms.models.pages.page.Page ]
+
+    :return: the depth of node whithin the tree/pages in pageset
+    :rtype: int
+    """
+    if not node.parent in pageset:
+        return 0
+    return node.depth - get_highest_anscentor_in(node, pageset).depth
```

Comment:
Really cool how you handled the depth calculation of partial trees :muscle: 
Per default only pages from level 0 (the "roots") are shown in the tree,
So we just needed this solution to decide, which pages are the "roots" in a partial tree